addons/api/addon/serializers/credential.js

```diff
@@ -13,6 +13,20 @@ export default class CredentialSerializer extends ApplicationSerializer {
    */
   serializeScopeID = false;
 
+  serialize() {
+    const serialized = super.serialize(...arguments);
+
+    if (serialized.type === 'json') {
+      delete serialized.attributes.username;
+
+      serialized.attributes.object = serialized.attributes.json_object;
+      delete serialized.attributes.json_object;
+      let parsedSecret = JSON.parse(serialized.attributes.object);
+      serialized.attributes.object = parsedSecret;
```

Comment:
```
      if (serialized?.attributes?.json_object) {
            const parsedSecret = JSON.parse(serialized.attributes.json_object);
            delete serialized.attributes.json_object;
            serialized.attributes.object = parsedSecret;
      }
```
I just remembered we need a check to make sure `json_object` has value before we try to parse it. In some cases like editing, they may not have updated the secret and the `json_object` key/value pair won't exist anymore.

Reply:
@cameronperera take a look when you get a chance